src/dmd/dtemplate.d

```diff
@@ -4324,6 +4324,14 @@ MATCH deduceType(RootObject o, Scope* sc, Type tparam, TemplateParameters* param
                 }
             }
 
+            bool isTopRef(Type t)
```

Comment:
Maybe just need a small comment on what this function returns.  True if the `t` is a reference type, or an array of reference types.